### PR TITLE
Make snapshot process snapshot all unflushed blocks

### PR DESCRIPTION
**Pull request by richardartoul**
- [X] - Modify snapshot behavior to snapshot all unflushed blocks within the retention period
- [X] - Update integration tests to capture new behavior

### CODE REVIEW

File: src/dbnode/storage/util.go [hunk 1]
```diff
@@ -49,7 +49,7 @@ func timesInRange(startInclusive, endInclusive time.Time, windowSize time.Durati
 		return nil
 	}
 	times := make([]time.Time, 0, ni)
-	for t := endInclusive; !t.Before(startInclusive); t = t.Add(-windowSize) {
```
Comment (justinjc) [inline src/dbnode/storage/util.go]: Were there assumptions elsewhere in the code that relies on the time ranges always including the end time? Also fix the comment for this function.

**Comment by prateek** [comment]
@richardartoul talked offline: maybe add a stat for the number of blocks being snapshot

File: src/dbnode/integration/disk_flush_helpers.go [hunk 2]
```diff
@@ -46,23 +46,59 @@ var (
...
+	namespace ident.ID,
+	blockStart time.Time,
+) int {
+	latestVolumeIndex := -1
+
+	for _, shard := range shardSet.AllIDs() {
+		snapshotFiles, err := fs.SnapshotFiles(
+			filePathPrefix, namespace, shard)
+		if err != nil {
+			panic(err)
```
Comment (prateek) [inline src/dbnode/integration/disk_flush_helpers.go]: don't panic here, propagate error

Comment (prateek) [reply src/dbnode/integration/disk_flush_helpers.go]: meh just saw its only for integration tests, w/e

File: src/dbnode/storage/flush.go [hunk 3]
```diff
@@ -233,6 +237,28 @@ func (m *flushManager) namespaceFlushTimes(ns databaseNamespace, curr time.Time)
...
+		// allow writes to be written into the next block. Note that "current time"
+		// here is defined as "tick start time" because all the guarantees about
+		// snapshotting are based around the tick start time, now the current time.
+		latest = curr.Add(rOpts.BufferFuture()).Truncate(blockSize)
+	)
+
+	candidateTimes := timesInRange(earliest, latest, blockSize)
+	return filterTimes(candidateTimes, func(t time.Time) bool {
+		// Snapshot anything that is unflushed.
+		return ns.NeedsFlush(t, t)
```
Comment (prateek) [inline src/dbnode/storage/flush.go]: do you need to check snapshot enabled here

**Pull request merged**